### PR TITLE
Journey page

### PR DESCRIPTION
**Pull request by iHiD**
This adds the start of the journey page, and as part of this adds the start of a dark theme.

Screenshot 2021-01-06 at 23 04 28
Screenshot 2021-01-06 at 23 04 36

---

Here are the light equivelents of these components:
Screenshot 2021-01-06 at 23 10 36

Screenshot 2021-01-06 at 23 10 45

---

The dark theme works as follows.

- In app/css/application.css we define all our colours with colour-based names (e.g. lightBlue, darkBlue, etc) - currently these are in the format of `--c-XXX000`, but I'll name them all properly as a future step.
- We define two classes: `theme-dark` and `theme-light` which set semantic colours to the color-named colours (e.g. `textColor5`, `inputBackgroundColor`). It is expected that multiple semantic colours will share the same colour-named color. Some colors follow a 1-9 pattern (e.g `textColor1`, `textColor5`, `textColor9`). In light-theme the lower the number, the darker the color (so at the extremities `textColor1` is black and `textColor9` is white).
- We have a tailwind helper defined for each semantic colour.
- In our CSS we `@apply textColor1`.
- By changing the `theme-light` to `theme-dark` on the body class, the whole site changes theme.
- By adding `theme-XYZ` to any element, that element and it's children adopt that style (for example, the main part of the Journey page should be dark-themed regardless of overall preference).

This is a transitional WIP. New CSS will be written with the new helpers, and old CSS will be migrated over January.

---

https://user-images.githubusercontent.com/286476/103832176-94a26b00-5075-11eb-806a-7d02b7ba055d.mov

### CODE REVIEW

File: app/css/application.css [hunk 1]
```diff
@@ -18,8 +18,61 @@
 @import "tailwindcss/components";
 @import "tailwindcss/utilities";
 
+:root {
+    /* TODO: Rename these to match figma */
+    --c-CBC9D9: #cbc9d9;
```
Comment (iHiD) [inline app/css/application.css:23]: These will be named to `darkBlue` etc. They cover the entire Exercism palette, both light and dark

File: app/css/application.css [hunk 2]
```diff
@@ -18,8 +18,61 @@
...
+    --c-26282D: #26282d;
+    --c-000000: #000000;
+    --c-FFFFFF: #ffffff;
+    --c-FBFCFE: #fbfcfe;
+    --c-282339: #282339;
+    --c-A9A6BD: #a9a6bd;
+}
+
+.theme-light {
+    --backgroundColorA: var(--c-FFFFFF);
```
Comment (iHiD) [inline app/css/application.css:45]: The semantic colours used within the code are then mapped to the named variables above.

File: app/css/components/search-bar.css [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    @apply flex items-center;
+
+    & > .container {
+        @apply flex items-center;
+    }
+
+    & input.--search {
+        @apply mr-24 rounded-5 border-2 border-transparent h-48;
+        @apply pl-40 pr-12;
+        @apply text-16 bg-inputBackgroundColor;
```
Comment (iHiD) [inline app/css/components/search-bar.css:12]: These semantic colours are then used throughout the rest of the CSS.

File: tailwind.config.js [hunk 4]
```diff
@@ -32,19 +32,36 @@ module.exports = {
...
+      textColor6: 'var(--textColor4)',
+      textColor9: 'var(--textColor9)',
+      inputBackgroundColor: 'var(--inputBackgroundColor)',
+      inputBorderColorFocus: 'var(--inputBorderColorFocus)',
+      backgroundColorA: 'var(--backgroundColorA)',
+      backgroundColorB: 'var(--backgroundColorB)',
+      backgroundColorC: 'var(--backgroundColorC)',
+      borderColor4: 'var(--borderColor4)',
+      borderColor5: 'var(--borderColor5)',
+      borderColor7: 'var(--borderColor7)',
```
Comment (iHiD) [inline tailwind.config.js:46]: New colours follow the pattern of: `{element}Color{variant}`

**Comment by iHiD** [comment]
https://user-images.githubusercontent.com/286476/103832176-94a26b00-5075-11eb-806a-7d02b7ba055d.mov

**Comment by taiyab** [comment]
This looks sick. Nice work @iHiD

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**